core/src/main/java/org/mapfish/print/map/geotools/grid/GridLayer.java

```diff
@@ -134,7 +121,26 @@ public void render(
     }
   }
 
-  private void applyOffset(final AffineTransform transform, final GridLabel.Side side) {
+  /**
+   * Loop through available Fonts until one supports the dpiScaling.
+   * @param dpiScaling used to calculate the font physical size.
+   * @return the first available Font or NPE if none exists
+   */
+    private Font getBaseFont(final double dpiScaling) {
+        for (String fontName : this.params.font.name) {
+          try {
+              return new Font(
+                  fontName,
+                  this.params.font.style.styleId,
+                  (int) (this.params.font.size * dpiScaling));
+          } catch (RuntimeException e) {
+            // try next font in list
+          }
+        }
+        throw new NullPointerException("Found no matching font for dpiScaling:" + dpiScaling);
```

Comment:
You just add a NullPointerException?
Why does it fix a SonarCloud issue?
Shouldn't be a RuntimeException

Reply:
New RuntimeException as discussed